### PR TITLE
Add integration test for revision-based upgrade

### PR DESCRIPTION
**Pull request by Monkeyanator**
Integration test for revision-based upgrades that checks that a rollout completed with updated images and that tests traffic before and after the rollout.

Later work will include making this test agnostic to the upgrade method (i.e. pass in an `UpgradeFunc`, doesn't matter if in-place or revisioned), sending traffic during upgrade, testing that metrics continued to accumate during the upgrade.

/cc @jacob-delgado

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[x] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by ericvn — APPROVED** [review]

**Comment by Monkeyanator** [comment]
Sorry, should have mentioned in a comment: I noticed a flake on this test in CI related to the rollout, going to try and reproduce before looking at merging

**Review by yangminzhu — APPROVED** [review]
Looks good for the change in tests/integration/security/authorization_test.go

**Review by jacob-delgado — APPROVED** [review]

File: tests/integration/pilot/revisioned_upgrade_test.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+// testUpgradeFromVersion tests an upgrade from the target version to the master version
+// provided fromVersion must be present in tests/integration/pilot/testdata/upgrade for the installation to succeed
+// TODO(monkeyanator) pass this a generic UpgradeFunc allowing for reuse across in-place and revisioned upgrades
+func testUpgradeFromVersion(ctx framework.TestContext, t *testing.T, fromVersion string) {
+	configs := make(map[string]string)
+	ctx.WhenDone(func() error {
+		var errs *multierror.Error
+		for _, config := range configs {
+			ctx.Config().DeleteYAML("istio-system", config)
+			multierror.Append(errs, ctx.Config().DeleteYAML("istio-system", config))
```
[the remark this answers is not in the document]
Comment (Monkeyanator) [reply tests/integration/pilot/revisioned_upgrade_test.go:59]: it should not be, good catch :+1: Fixed in latest

**Review by howardjohn — APPROVED** [review]

**Pull request merged**